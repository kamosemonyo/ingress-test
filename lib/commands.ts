export class CommonCommands {
  static installCdkCmd: string[] = [
    'echo "Installing AWS CDK"',
    'npm install -g aws-cdk typescript',
    'cdk --version',
  ];

  static debugEnvCmd = [
    'env',
    'ls -al',
  ];
};
