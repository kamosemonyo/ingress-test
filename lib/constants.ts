import { LinuxBuildImage } from '@aws-cdk/aws-codebuild';

export const codeBuildSpecVersion = '0.2';
export const defaultCodeBuildEnvironment = { buildImage: LinuxBuildImage.STANDARD_5_0 };