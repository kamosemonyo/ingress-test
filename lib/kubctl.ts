
export class Kubectl {

  static login (clusterName:string, region?:string) {
    const clusterRegion = (region == undefined) ? 'af-south-1': region
    return `aws eks update-kubeconfig --name ${clusterName} --region ${clusterRegion}`
  }

  static applyFolder (folder:string) {
    return `kubectl apply -f ${folder}`
  }
}