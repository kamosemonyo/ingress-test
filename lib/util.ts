import { CfnParameter } from "@aws-cdk/core";

export const toValidConstructName = (id: string) => {
  let str = id.split('_').join('-');

  return str.split('-')
  .map(w => w[0].toUpperCase() + w.substr(1).toLowerCase())
  .join('-');
};
