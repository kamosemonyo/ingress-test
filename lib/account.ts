import { ACCOUNT_PRE, ACCOUNT_PROD, ECR_REGION, EKS_REGION, ENV_PRE, ENV_PROD, PIPELINE_REGION } from "./constants";

export class Account {
  

  static forPipeline (env:string = ENV_PRE) {
    if (env !== ENV_PRE && env !== ENV_PROD) {
      env = 'pre';
    }

    if (env == ENV_PROD) {
      return {
        region: PIPELINE_REGION,
        account: ACCOUNT_PROD,
      };
    }

    return {
      region: PIPELINE_REGION,
      account: ACCOUNT_PRE,
    };
  }

  static forCluster (env:string = ENV_PRE) {
    if (env !== ENV_PRE && env !== ENV_PROD) {
      env = 'pre';
    }

    if (env == ENV_PROD) {
      return {
        region: EKS_REGION,
        account: ACCOUNT_PROD,
      };
    }

    return {
      region: EKS_REGION,
      account: ACCOUNT_PRE,
    };
  }

  static forImages (env:string = ENV_PRE) {
    if (env !== ENV_PRE && env !== ENV_PROD) {
      env = 'pre';
    }

    if (env == ENV_PROD) {
      return {
        region: ECR_REGION,
        account: ACCOUNT_PROD,
      };
    }

    return {
      region: ECR_REGION,
      account: ACCOUNT_PRE,
    };
  }
}