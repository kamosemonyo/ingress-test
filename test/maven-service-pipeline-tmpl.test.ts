import { SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';

import { MavenServicePipeline } from '../templates/maven-service-pipeline';

import '@aws-cdk/assert/jest';
import { toValidConstructName } from '../lib/util';

const repositoryName = 'pfm-service';
const artifactsBucket = 'pipeline-artifacts';

test('Stack Snapshot', () => {
  const app = new cdk.App({ context: { repositoryName }});

  const stack = new MavenServicePipeline(app, toValidConstructName(repositoryName), {});

  app.synth();

  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});

test('Throw error without repository context', () => {
  const synth = () => {
    const app = new cdk.App();
    const stack = new MavenServicePipeline(app, toValidConstructName(repositoryName), {});
    app.synth();
  };

  expect(synth).toThrow('Expected context [repositoryName] to be defined.');
});

test('Creates all required resources', () => {
  const app = new cdk.App({ context: { repositoryName }});
  const stack = new MavenServicePipeline(app, toValidConstructName(repositoryName), {});
  app.synth();

  // Creates Pipeline
  expect(stack).toHaveResource('AWS::CodePipeline::Pipeline', {
    Name: {
      'Fn::Join': [
        '',
        [
          `${repositoryName}-`,
          {
            'Ref': 'branch'
          }
        ]
      ]
    },
    ArtifactStore: {
      Location: artifactsBucket,
      Type: 'S3'
    },
  });

  // Creates build CodeBuild project
  expect(stack).toHaveResource('AWS::CodeBuild::Project', {
    Name: {
      'Fn::Join': [
        '',
        [
          `${repositoryName}-`,
          {
            'Ref': 'branch',
          },
          '-build',
        ],
      ],
    }
  });

  // Creates deploy CodeBuild project
  expect(stack).toHaveResource('AWS::CodeBuild::Project', {
    Name: {
      'Fn::Join': [
        '',
        [
          `${repositoryName}-`,
          {
            'Ref': 'branch',
          },
          '-deploy',
        ],
      ],
    }
  });

  // Creates build ECR repository
  expect(stack).toHaveResource('AWS::ECR::Repository', {
    RepositoryName: repositoryName,
    ImageTagMutability: 'IMMUTABLE'
  });

  // // Creates build CB project
  // expect(stack).toHaveResource('', {

  // });
});

test('Builds pipeline correctly', () => {
  process.env.SOURCE_BRANCH = 'develop';

  const app = new cdk.App({ context: { repositoryName }});
  const stack = new MavenServicePipeline(app, toValidConstructName(repositoryName), {});

  app.synth();

  expect(stack).toHaveResourceLike('AWS::CodePipeline::Pipeline', {
    Name: {
      'Fn::Join': [
        '',
        [
          `${repositoryName}-`,
          {
            'Ref': 'branch'
          }
        ]
      ]
    },
    Stages: [
      {
        Actions: [
          {
            ActionTypeId: {
              Category: 'Source',
              Provider: 'GitHub',
            },
            Configuration: {
              Repo: repositoryName,
              Branch: 'master',
              OAuthToken: '{{resolve:secretsmanager:github-mmi-holdings-ces-token:SecretString:::}}',
            },
            OutputArtifacts: [
              {
                Name: 'Artifact_Source_Github_Source'
              }
            ],
          }
        ],
        Name: 'Source'
      },
      {
        Actions: [
          {
            ActionTypeId: {
              Category: 'Approval',
              Provider: 'Manual',
            },
          }
        ],
        Name: 'Approve_Build'
      },
      {
        Actions: [
          {
            ActionTypeId: {
              Provider: 'CodeBuild',
            },
            InputArtifacts: [
              {
                Name: 'Artifact_Source_Github_Source'
              }
            ],
            Name: 'Maven_Docker_Build',
            OutputArtifacts: [
              {
                Name: 'Artifact_Build_Maven_Docker_Build'
              }
            ],
          }
        ],
        Name: 'Build'
      },
      {
        Actions: [
          {
            ActionTypeId: {
              Category: 'Approval',
              Provider: 'Manual',
            },
            Name: 'Approve',
          }
        ],
        Name: 'Approve_Deploy'
      },
      {
        Actions: [
          {
            ActionTypeId: {
              Provider: 'CodeBuild',
            },
            InputArtifacts: [
              {
                Name: 'Artifact_Build_Maven_Docker_Build'
              }
            ],
            Name: 'Ecs_Deploy',
          }
        ],
        Name: 'Deploy'
      }
    ]
  });

});
