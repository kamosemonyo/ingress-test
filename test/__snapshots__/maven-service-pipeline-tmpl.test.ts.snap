// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack Snapshot 1`] = `
Object {
  "Parameters": Object {
    "branch": Object {
      "Default": "master",
      "Description": "Github/Codecommit source branch",
      "Type": "String",
    },
    "githuborg": Object {
      "Default": "mmi-holdings-ces",
      "Description": "Github organisation name",
      "Type": "String",
    },
    "pomfile": Object {
      "AllowedPattern": "*pom.xml",
      "Default": "pom.xml",
      "Description": "pom.xml file path",
      "Type": "String",
    },
    "propertiesfile": Object {
      "Description": "pom.properties file path",
      "Type": "String",
    },
    "sourceprovider": Object {
      "AllowedValues": Array [
        "github",
        "codecommit",
        "s3",
      ],
      "Default": "github",
      "Description": "Pipeline source provider",
      "Type": "String",
    },
  },
  "Resources": Object {
    "PfmServiceCodeBuildEcsDeployProjectC6FDDE22": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "CODEPIPELINE",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "Name": Object {
          "Fn::Join": Array [
            "",
            Array [
              "pfm-service-",
              Object {
                "Ref": "branch",
              },
              "-deploy",
            ],
          ],
        },
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "PfmServiceCodeBuildEcsDeployProjectRole648B05E0",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": "{
  \\"version\\": \\"0.2\\",
  \\"phases\\": {
    \\"install\\": {
      \\"commands\\": [
        \\"env\\",
        \\"ls -al\\",
        \\"echo \\\\\\"Installing AWS CDK\\\\\\"\\",
        \\"npm install -g aws-cdk typescript\\",
        \\"cdk --version\\"
      ]
    },
    \\"build\\": {
      \\"commands\\": []
    }
  }
}",
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "PfmServiceCodeBuildEcsDeployProjectRole648B05E0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PfmServiceCodeBuildEcsDeployProjectRoleDefaultPolicy23D57A21": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PfmServiceCodeBuildEcsDeployProjectC6FDDE22",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PfmServiceCodeBuildEcsDeployProjectC6FDDE22",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "PfmServiceCodeBuildEcsDeployProjectC6FDDE22",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::pipeline-artifacts",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::pipeline-artifacts/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PfmServiceCodeBuildEcsDeployProjectRoleDefaultPolicy23D57A21",
        "Roles": Array [
          Object {
            "Ref": "PfmServiceCodeBuildEcsDeployProjectRole648B05E0",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PfmServiceCodeBuildProject660F0741": Object {
      "Properties": Object {
        "Artifacts": Object {
          "Type": "CODEPIPELINE",
        },
        "EncryptionKey": "alias/aws/s3",
        "Environment": Object {
          "ComputeType": "BUILD_GENERAL1_SMALL",
          "Image": "aws/codebuild/standard:5.0",
          "ImagePullCredentialsType": "CODEBUILD",
          "PrivilegedMode": false,
          "Type": "LINUX_CONTAINER",
        },
        "Name": Object {
          "Fn::Join": Array [
            "",
            Array [
              "pfm-service-",
              Object {
                "Ref": "branch",
              },
              "-build",
            ],
          ],
        },
        "ServiceRole": Object {
          "Fn::GetAtt": Array [
            "PfmServiceCodeBuildProjectRole4E78BE74",
            "Arn",
          ],
        },
        "Source": Object {
          "BuildSpec": Object {
            "Fn::Join": Array [
              "",
              Array [
                "{
  \\"version\\": \\"0.2\\",
  \\"phases\\": {
    \\"install\\": {
      \\"runtime_versions\\": {
        \\"java\\": \\"corretto8\\"
      }
    },
    \\"build\\": {
      \\"commands\\": [
        \\"java -version\\",
        \\"mvn -version\\",
        \\"mvn clean install\\",
        \\"cp ",
                Object {
                  "Ref": "propertiesfile",
                },
                " docker/files\\",
        \\"export VERSION=\`grep -oP 'version=\\\\\\\\K.*' ",
                Object {
                  "Ref": "propertiesfile",
                },
                "\`\\",
        \\"echo $VERSION > VERSION\\",
        \\"docker build -t ",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ".dkr.ecr.",
                Object {
                  "Ref": "AWS::Region",
                },
                ".amazonaws.com/pfm-service:$VERSION .\\",
        \\"aws ecr get-login-password --region ",
                Object {
                  "Ref": "AWS::Region",
                },
                " | docker login --username AWS --password-stdin ",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ".dkr.ecr.",
                Object {
                  "Ref": "AWS::Region",
                },
                ".amazonaws.com\\",
        \\"docker push ",
                Object {
                  "Ref": "AWS::AccountId",
                },
                ".dkr.ecr.",
                Object {
                  "Ref": "AWS::Region",
                },
                ".amazonaws.com/pfm-service:$VERSION\\"
      ]
    }
  },
  \\"artifacts\\": {
    \\"files\\": [
      \\"VERSION\\",
      \\"",
                Object {
                  "Ref": "propertiesfile",
                },
                "\\"
    ]
  }
}",
              ],
            ],
          },
          "Type": "CODEPIPELINE",
        },
      },
      "Type": "AWS::CodeBuild::Project",
    },
    "PfmServiceCodeBuildProjectRole4E78BE74": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codebuild.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PfmServiceCodeBuildProjectRoleDefaultPolicy27285B03": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "logs:CreateLogGroup",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PfmServiceCodeBuildProject660F0741",
                      },
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":logs:",
                      Object {
                        "Ref": "AWS::Region",
                      },
                      ":",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":log-group:/aws/codebuild/",
                      Object {
                        "Ref": "PfmServiceCodeBuildProject660F0741",
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": Array [
                "codebuild:CreateReportGroup",
                "codebuild:CreateReport",
                "codebuild:UpdateReport",
                "codebuild:BatchPutTestCases",
                "codebuild:BatchPutCodeCoverages",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    "arn:",
                    Object {
                      "Ref": "AWS::Partition",
                    },
                    ":codebuild:",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ":report-group/",
                    Object {
                      "Ref": "PfmServiceCodeBuildProject660F0741",
                    },
                    "-*",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "ecr:BatchGetImage",
                "ecr:BatchCheckLayerAvailability",
                "ecr:GetDownloadUrlForLayer",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::Join": Array [
                  "",
                  Array [
                    Object {
                      "Ref": "AWS::AccountId",
                    },
                    ".dkr.ecr.",
                    Object {
                      "Ref": "AWS::Region",
                    },
                    ".amazonaws.com/pfm-service",
                  ],
                ],
              },
            },
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::pipeline-artifacts",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::pipeline-artifacts/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PfmServiceCodeBuildProjectRoleDefaultPolicy27285B03",
        "Roles": Array [
          Object {
            "Ref": "PfmServiceCodeBuildProjectRole4E78BE74",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PfmServiceEcrRepository23177656": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ImageScanningConfiguration": Object {
          "ScanOnPush": true,
        },
        "ImageTagMutability": "IMMUTABLE",
        "LifecyclePolicy": Object {
          "LifecyclePolicyText": "{\\"rules\\":[{\\"rulePriority\\":1,\\"selection\\":{\\"tagStatus\\":\\"tagged\\",\\"tagPrefixList\\":[\\"SNAPSHOT\\"],\\"countType\\":\\"imageCountMoreThan\\",\\"countNumber\\":40},\\"action\\":{\\"type\\":\\"expire\\"}},{\\"rulePriority\\":2,\\"selection\\":{\\"tagStatus\\":\\"any\\",\\"countType\\":\\"imageCountMoreThan\\",\\"countNumber\\":9999},\\"action\\":{\\"type\\":\\"expire\\"}}]}",
        },
        "RepositoryName": "pfm-service",
      },
      "Type": "AWS::ECR::Repository",
      "UpdateReplacePolicy": "Retain",
    },
    "PfmServicePipeline8BC7E9B0": Object {
      "DependsOn": Array [
        "PfmServicePipelineRoleDefaultPolicy5ED3348A",
        "PfmServicePipelineRoleEB6A1491",
      ],
      "Properties": Object {
        "ArtifactStore": Object {
          "Location": "pipeline-artifacts",
          "Type": "S3",
        },
        "Name": Object {
          "Fn::Join": Array [
            "",
            Array [
              "pfm-service-",
              Object {
                "Ref": "branch",
              },
            ],
          ],
        },
        "RoleArn": Object {
          "Fn::GetAtt": Array [
            "PfmServicePipelineRoleEB6A1491",
            "Arn",
          ],
        },
        "Stages": Array [
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Source",
                  "Owner": "ThirdParty",
                  "Provider": "GitHub",
                  "Version": "1",
                },
                "Configuration": Object {
                  "Branch": "master",
                  "OAuthToken": "{{resolve:secretsmanager:github-mmi-holdings-ces-token:SecretString:::}}",
                  "Owner": Object {
                    "Ref": "githuborg",
                  },
                  "PollForSourceChanges": false,
                  "Repo": "pfm-service",
                },
                "Name": "Github_Source",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "Artifact_Source_Github_Source",
                  },
                ],
                "RunOrder": 1,
              },
            ],
            "Name": "Source",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Approval",
                  "Owner": "AWS",
                  "Provider": "Manual",
                  "Version": "1",
                },
                "Name": "Approve",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PfmServicePipelineApproveBuildApproveCodePipelineActionRoleEB826E78",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Approve_Build",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ProjectName": Object {
                    "Ref": "PfmServiceCodeBuildProject660F0741",
                  },
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "Artifact_Source_Github_Source",
                  },
                ],
                "Name": "Maven_Docker_Build",
                "OutputArtifacts": Array [
                  Object {
                    "Name": "Artifact_Build_Maven_Docker_Build",
                  },
                ],
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PfmServicePipelineBuildMavenDockerBuildCodePipelineActionRole660E94A1",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Build",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Approval",
                  "Owner": "AWS",
                  "Provider": "Manual",
                  "Version": "1",
                },
                "Name": "Approve",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PfmServicePipelineApproveDeployApproveCodePipelineActionRoleE18751FB",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Approve_Deploy",
          },
          Object {
            "Actions": Array [
              Object {
                "ActionTypeId": Object {
                  "Category": "Build",
                  "Owner": "AWS",
                  "Provider": "CodeBuild",
                  "Version": "1",
                },
                "Configuration": Object {
                  "ProjectName": Object {
                    "Ref": "PfmServiceCodeBuildEcsDeployProjectC6FDDE22",
                  },
                },
                "InputArtifacts": Array [
                  Object {
                    "Name": "Artifact_Build_Maven_Docker_Build",
                  },
                ],
                "Name": "Ecs_Deploy",
                "RoleArn": Object {
                  "Fn::GetAtt": Array [
                    "PfmServicePipelineDeployEcsDeployCodePipelineActionRoleD714E0C9",
                    "Arn",
                  ],
                },
                "RunOrder": 1,
              },
            ],
            "Name": "Deploy",
          },
        ],
      },
      "Type": "AWS::CodePipeline::Pipeline",
    },
    "PfmServicePipelineApproveBuildApproveCodePipelineActionRoleEB826E78": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PfmServicePipelineApproveDeployApproveCodePipelineActionRoleE18751FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PfmServicePipelineBuildMavenDockerBuildCodePipelineActionRole660E94A1": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PfmServicePipelineBuildMavenDockerBuildCodePipelineActionRoleDefaultPolicy7D9B93B0": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PfmServiceCodeBuildProject660F0741",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PfmServicePipelineBuildMavenDockerBuildCodePipelineActionRoleDefaultPolicy7D9B93B0",
        "Roles": Array [
          Object {
            "Ref": "PfmServicePipelineBuildMavenDockerBuildCodePipelineActionRole660E94A1",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PfmServicePipelineDeployEcsDeployCodePipelineActionRoleD714E0C9": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PfmServicePipelineDeployEcsDeployCodePipelineActionRoleDefaultPolicy8B3E5F71": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "codebuild:BatchGetBuilds",
                "codebuild:StartBuild",
                "codebuild:StopBuild",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PfmServiceCodeBuildEcsDeployProjectC6FDDE22",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PfmServicePipelineDeployEcsDeployCodePipelineActionRoleDefaultPolicy8B3E5F71",
        "Roles": Array [
          Object {
            "Ref": "PfmServicePipelineDeployEcsDeployCodePipelineActionRoleD714E0C9",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PfmServicePipelineRoleDefaultPolicy5ED3348A": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::pipeline-artifacts",
                    ],
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::pipeline-artifacts/*",
                    ],
                  ],
                },
              ],
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PfmServicePipelineApproveBuildApproveCodePipelineActionRoleEB826E78",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PfmServicePipelineBuildMavenDockerBuildCodePipelineActionRole660E94A1",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PfmServicePipelineApproveDeployApproveCodePipelineActionRoleE18751FB",
                  "Arn",
                ],
              },
            },
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "PfmServicePipelineDeployEcsDeployCodePipelineActionRoleD714E0C9",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "PfmServicePipelineRoleDefaultPolicy5ED3348A",
        "Roles": Array [
          Object {
            "Ref": "PfmServicePipelineRoleEB6A1491",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "PfmServicePipelineRoleEB6A1491": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "codepipeline.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "PfmServicePipelineSourceGithubSourceWebhookResource6916ABDA": Object {
      "Properties": Object {
        "Authentication": "GITHUB_HMAC",
        "AuthenticationConfiguration": Object {
          "SecretToken": "{{resolve:secretsmanager:github-mmi-holdings-ces-token:SecretString:::}}",
        },
        "Filters": Array [
          Object {
            "JsonPath": "$.ref",
            "MatchEquals": "refs/heads/{Branch}",
          },
        ],
        "RegisterWithThirdParty": true,
        "TargetAction": "Github_Source",
        "TargetPipeline": Object {
          "Ref": "PfmServicePipeline8BC7E9B0",
        },
        "TargetPipelineVersion": 1,
      },
      "Type": "AWS::CodePipeline::Webhook",
    },
  },
}
`;
